### PR TITLE
Move ShouldHandleFileOpEvent unit tests into their own class

### LINKED ISSUE #1195: Move ShouldHandleFileOpEvent tests into their own class
Update the kext unit tests based on the comments left by @pmj in #1194 

In addition to moving `ShouldHandleFileOpEvent` tests, also:

> I realise the old code did this too, but the #defined name, PrjFSProviderUserClient is generally preferable in code - that's the whole point of the macro, if the reverse DNS prefix changes, you don't have to update all of the code.
>
>Might be worth doing a search for the use of the explicitly-prefixed name, I notice quite a few instances of it being used in this file.

### PR DESCRIPTION
**Pull request by wilbaker**
Additionally:
- Reorder PrjFSKextTests files so they're in alphabetical order
- Replace the usage of org_vfsforgit_PrjFSProviderUserClient with PrjFSProviderUserClient

Follow up work for comments left on #1194 after it was merged.

Resolves #1195

### CODE REVIEW

File: ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+#include "MockProc.hpp"
+#include "MockVnodeAndMount.hpp"
+#include "VnodeCacheEntriesWrapper.hpp"
+
+#import <XCTest/XCTest.h>
+
+using std::shared_ptr;
+
+// Dummy implementation
+class org_vfsforgit_PrjFSProviderUserClient
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: This is still using the reverse-DNS prefixed name.

File: ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            &self->perfTracer,
+            self->context,
+            self->repoRootVnode.get(),
+            nullptr, // path
+            KAUTH_FILEOP_RENAME,
+            true, // isDirectory,
+            &testRootHandle,
+            &pid));
+
+    PrjFSProviderUserClient userClient;
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: There's probably no harm moving this to an ivar - various other tests also use one of these.

File: ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        ShouldHandleFileOpEvent(
+            &self->perfTracer,
+            self->context,
+            self->testVnodeFile.get(),
+            nullptr, // path
+            KAUTH_FILEOP_OPEN,
+            false, // isDirectory,
+            &rootHandle,
+            &pid));
+    XCTAssertTrue(rootHandle == testRootHandle);
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: While we're poking around in here, would it be worth changing these assertions to `XCTAssertEqual`?

File: ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        0,
+        self->repoRootVnode.get(),
+        self->repoRootVnode->GetVid(),
+        FsidInode{ self->repoRootVnode->GetMountPoint()->GetFsid(), self->repoRootVnode->GetInode() },
+        self->repoPath.c_str());
+    XCTAssertTrue(VirtualizationRoot_IsValidRootHandle(testRootHandle));
+
+    int pid;
+    // Fail when the provider is not online
+    s_virtualizationRoots[0].providerUserClient = nullptr;
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: This should really be an `ActiveProvider_Disconnect()` call. (In which case there needs to be a `vnode_get(self->repoRootVnode.get())` after the insertion above to balance the `vnode_put` in Disconnect; this is a bit ugly and I believe I fixed the insertion to perform the `_get` in #500. I think I'll extract the simpler bug fixes from that PR and resubmit them as a standalone one.)

Comment (wilbaker) [reply ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: I see that you made this change in #1200.  

I will leave this as-is here and wait to merge this PR in until after you've merged #1200.

**Comment by wilbaker** [comment]
Waiting on #1200 to be merged before merging this PR.

File: ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm (hunk 4 above)
Comment (wilbaker) [reply ProjFS.Mac/PrjFSKextTests/ShouldHandleFileOpTests.mm]: Actually I see that this is a slightly different suggestion, fixing now

**Pull request merged**